packages/client/src/StreamVideoClient.ts

```diff
@@ -69,11 +72,19 @@ export class StreamVideoClient {
    * @param tokenOrProvider a token or a function that returns a token.
    */
   connectUser = async (user: User, tokenOrProvider: TokenOrProvider) => {
-    const connectUserResponse = await this.streamClient.connectUser(
-      // @ts-expect-error
-      user,
-      tokenOrProvider,
-    );
+    const connectUser = () => {
+      return this.streamClient.connectUser(
+        // @ts-expect-error
+        user,
+        tokenOrProvider,
+      );
+    };
+    this.connectionPromise = this.disconnectionPromise
+      ? this.disconnectionPromise.then(() => connectUser())
+      : connectUser();
+
+    this.connectionPromise?.finally(() => (this.connectionPromise = undefined));
+    const connectUserResponse = await this.connectionPromise;
```

Comment:
Do we need to handle this here? I'd probably leave this up to the integrators... In our hook, we had to deal with this because of React's StrictMode where every effect is executed twice in dev mode:
- https://react.dev/reference/react/StrictMode